Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelper.java

```diff
@@ -0,0 +1,202 @@
+
+        assertEquals("DocumentQueryOut", actualCheckPolicy.getRequest().getAction().getAttribute().get(0)
+            .getAttributeValue().get(0).getContent().get(0));
+        SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+        AttributeType attributeType = subjectType.getAttribute().get(1);
+        assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+        assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+        assertEquals("urn:oid:1.1.1.1.1.1.1", attributeType.getAttributeValue().get(0).getContent().get(0));
+    }
+  
+
+    private void verifycheckInboundPolicy(CheckPolicyRequestType actualCheckPolicy) {
+        // Verify result
+        assertNotNull(actualCheckPolicy);
+        // AssertAssertion to make sure nothing change
+        AssertionType assertTypeResult = actualCheckPolicy.getAssertion();
+
+        assertEquals("messageId should not change", "12345566", assertTypeResult.getMessageId());
+        // Assert Resource Session
+        assertEquals("urn:oasis:names:tc:xacml:1.0:resource:resource-id", actualCheckPolicy.getRequest().getResource()
+            .get(0).getAttribute().get(1).getAttributeId());
+        String expectResourceId = "amp;26.489.22";
+        assertEquals(expectResourceId, actualCheckPolicy.getRequest().getResource().get(0).getAttribute().get(0)
+            .getAttributeValue().get(0).getContent().get(0));
+
+        assertEquals("DocumentQueryIn", actualCheckPolicy.getRequest().getAction().getAttribute().get(0)
+            .getAttributeValue().get(0).getContent().get(0));
+        SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+        AttributeType attributeType = subjectType.getAttribute().get(1);
+        assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+        assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+        assertEquals("urn:oid:1.1.1.1.1.1.1", attributeType.getAttributeValue().get(0).getContent().get(0));
+    }
+    
+ 
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_InValidDirection() {
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+        eventA.setDirection("in_outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNull("We should expect null checkPolicyEngine ", result);
+        eventA.setDirection(null);
+        assertNull(instance.transformAdhocQueryToCheckPolicy(eventA));
+        assertNull(instance.transformAdhocQueryToCheckPolicy(null));
+    }
+    
+ 
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_outbound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
```

Comment:
Agree with Tabassum or at least rename test method to something like this "testTransformAdhocQueryToCheckPolicy_DefaultOutBound"